manifest-updater/secrets/batchSigning.go

```diff
@@ -0,0 +1,101 @@
+package secrets
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+const (
+	batchSigningKeyExpirationDuration = 90 * 24 * time.Hour
+	// TODO: This needs to change to something like 70 days in prod
+	batchSigningKeyMaxAge = 20 * time.Second
```

Comment:
There's some problems with making this configurable because then we're going to start getting into conflict with the expiration duration. I want to rotate earlier than the expiration, but if I rotate too early then we might have already deleted a key that we considered "valid".

And on the point of configuration, how do you want the config to pass into the manifest updater? Do we introduce a new field in the CRD that defines the expiration date for these? What do we do if the CRD updates with a shorter expiration date, which expiration date do we recognize to be valid? Especially if we've advertised the other key before.

There's a lot of questions with making this configurable, we should probably write out the various scenarios before writing the code for it.